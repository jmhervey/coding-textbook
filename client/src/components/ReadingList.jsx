import React from 'react';
import styled from 'styled-components';

const StyledListHeader = styled.ul`
  font-size: 1.5rem;
  border-bottom: 1px solid gray;
  margin: 0.5rem;
  padding: 0.5rem;
`

const StyledList = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 1rem;
`
/*
 * Component for sidebar showing reading list / navigation 
 */
const ReadingList = React.memo( props => {

  const handleClick = (item) => {
    if(props.onSelect) props.onSelect(item);
  }

  const items = props.items.map(item => (
    <li key={item.id} onClick={()=>handleClick(item)}>
      {item.read ? "☑;" : "☐"}
      {item.title}
    </li>
  ))
    
  function openPopup() {
    window.open('https://www.google.com', 'Popup', 'width=400,height=300');
  }

  return (
    <nav>
      <StyledListHeader>Assigned Reading</StyledListHeader>
      <p>Click the below textbook section to open its context:</p>
      <StyledList>
        {items}
      </StyledList>
      <a href="#" class="hover-link">Help</a>
    </nav>
  )

})

export default ReadingList;