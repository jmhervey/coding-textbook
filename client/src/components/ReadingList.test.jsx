import React from 'react';
import { render } from '@testing-library/react';
import ReadingList from './ReadingList';
import fetchMock from "jest-fetch-mock";

// Configure fetch mock
fetchMock.enableMocks();

describe('ReadingList', () => {
  test('Unit Tests for the ReadingList component', () => {

    const items = [
      { id: 1, title: 'Article 1', content: 'Content 1' },
      { id: 2, title: 'Article 2', content: 'Content 2' },
    ];

    // Render the ReadingList component
    const { container } = render(<ReadingList items={items} />);

    // Test if the rendered component matches the saved snapshot
    expect(container.firstChild).toMatchSnapshot();
  });
});
