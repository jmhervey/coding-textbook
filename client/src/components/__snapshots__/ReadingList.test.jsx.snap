// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingList renders the ReadingList component 1`] = `
<nav>
  <ul
    class="sc-bcXHqe ivZAkS"
  >
    Textbooks
  </ul>
  <ul
    class="sc-gswNZR heYmLR"
  >
    <li>
      ☐
      Article 1
    </li>
    <li>
      ☐
      Article 2
    </li>
  </ul>
</nav>
`;
